.github/workflows/cibuild.yml

```diff
@@ -0,0 +1,108 @@
+
+jobs:
+  egginfo:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install tox
+        run: python -m pip install --upgrade tox
+      - name: egg-info
+        run: tox -e egg_info
+  tests:
+    needs: egginfo
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: [3.7, 3.8, 3.9]
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install tox
+      run: python -m pip install --upgrade tox
+    - name: Python 3.7 with latest astropy
+      if: matrix.python-version == 3.7
+      run: tox -e py37-test
+    - name: Python 3.7 with astropy30
+      if: matrix.python-version == 3.7
+      run: tox -e py37-test-astropy30
+    - name: Python 3.7 with astropy31
+      if: matrix.python-version == 3.7
+      run: tox -e py37-test-astropy31
+    - name: Python 3.7 with astropy32
+      if: matrix.python-version == 3.7
+      run: tox -e py37-test-astropy32
+    - name: Python 3.8 with latest astropy
+      if: matrix.python-version == 3.8
+      run: tox -e py38-test
+    - name: Python 3.8 with astropy32
+      if: matrix.python-version == 3.8
+      run: tox -e py38-test-astropy32
+    - name: Python 3.9 with latest astropy
+      if: matrix.python-version == 3.9
+      run: tox -e py39-test
+   
+  style_docs_coverage:
+    runs-on: ubuntu-latest
+    needs: tests
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Setup Graphviz
+        uses: ts-graphviz/setup-graphviz@v1
+      - name: Install tox
+        run: python -m pip install --upgrade tox
+      - name: Check codestyle
+        run: tox -e codestyle
+      - name: Check docs
+        run: tox -e docs
+      - name: Check coverage
+        run: tox -e cov
+      - name: Upload coverage to codecov
+        uses: codecov/codecov-action@v1
+        with:
+          file: ./coverage.xml
+  
+  astropydev:
+    # TODO: this job should be allowed to fail except that the functionality
+    # is not available in actions yet.
+    runs-on: ubuntu-latest
+    needs: tests
+    steps: 
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install tox
+        run: python -m pip install --upgrade tox
+      - name: Run tests against astropy dev
+        run: tox -e astropydev
+
+  #publish:
+  #needs: coverage
+  #TODO - should be trigger by tagging associated with creating a new release in
```

Comment:
Does this mean we won't have coverage per PR anymore?

Reply:
If you refer to the `needs: coverage` line this is just how the dependencies are expressed, e.g. coverage needs to successfully run first. As a matter of fact, coverage is only needed for PRs and it should be run for new releases. Something to fine tune later maybe.